src/coreclr/tools/Common/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1085,6 +1085,8 @@ protected virtual void EmitElementCount(ILCodeStream codeStream, MarshalDirectio
                     // if neither sizeConst or sizeParamIndex are specified, default to 1
                     codeStream.EmitLdc(1);
                 }
+
+                codeStream.Emit(ILOpcode.conv_ovf_i4);
```

Comment:
I think this can go after line 1080. All the other codepaths already produce things typed as int32.